gemp-swccg-logic/src/main/java/com/gempukku/swccgo/logic/effects/DeionizeStarshipEffect.java

```diff
@@ -0,0 +1,73 @@
+package com.gempukku.swccgo.logic.effects;
+
+import com.gempukku.swccgo.common.IonizationType;
+import com.gempukku.swccgo.game.PhysicalCard;
+import com.gempukku.swccgo.game.SwccgGame;
+import com.gempukku.swccgo.game.state.GameState;
+import com.gempukku.swccgo.logic.actions.SubAction;
+
+import com.gempukku.swccgo.logic.timing.AbstractSubActionEffect;
+import com.gempukku.swccgo.logic.timing.Action;
+import com.gempukku.swccgo.logic.timing.PassthruEffect;
+
+
+/**
+ * An effect to make a starship no longer Ionized.
+ */
+public class DeionizeStarshipEffect extends AbstractSubActionEffect {
+    private PhysicalCard _cardDeionized;
+    private boolean _deionizePower;
+    private boolean _deionizeDefense;
+    private boolean _deionizeHyperspeed;
+
+    /**
+     * Creates an effect to make a character no longer ionized.
+     * @param action the action performing this effect
+     * @param cardDeionized the card that is deionized
+     */
+    public DeionizeStarshipEffect(Action action, PhysicalCard cardDeionized, boolean deionizePower, boolean deionizeDefense, boolean deionizeHyperspeed) {
```

Comment:
Is there a purpose for having these multiple booleans?

Reply:
Mostly future-proofing if ionization were to be explored with future cards. Since we're applying different debuffs from being ionized (Planetary Cannon vs. starship Ion Cannons), it made sense to also setup the infrastructure for removing different as well.